src/library-authoring/create-collection/CreateCollectionModal.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {
+  ActionRow,
+  Button,
+  Form,
+  ModalDialog,
+} from '@openedx/paragon';
+import { useParams } from 'react-router-dom';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import FormikControl from '../../generic/FormikControl';
+import { LibraryContext } from '../common/context';
+import messages from './messages';
+import { useCreateLibraryCollection } from '../data/apiHooks';
+import { ToastContext } from '../../generic/toast-context';
+
+const CreateCollectionModal = () => {
+  const intl = useIntl();
+  const { libraryId } = useParams();
+  const create = useCreateLibraryCollection(libraryId);
+  const {
+    isCreateCollectionModalOpen,
+    closeCreateCollectionModal,
+  } = React.useContext(LibraryContext);
+  const { showToast } = React.useContext(ToastContext);
+
+  const [isCreatingCollection, setIsCreatingCollection] = React.useState<boolean>(false);
+
+  const handleOnClose = React.useCallback(() => {
+    closeCreateCollectionModal();
+    setIsCreatingCollection(false);
+  }, []);
+
+  const handleCreate = React.useCallback((values) => {
+    setIsCreatingCollection(true);
+
+    create.mutateAsync(values).then(() => {
+      handleOnClose();
+      showToast(intl.formatMessage(messages.createCollectionSuccess));
+    }).catch(() => {
+      setIsCreatingCollection(false);
+      showToast(intl.formatMessage(messages.createCollectionError));
+    });
+  }, []);
```

Comment:
If the submit function returns a `Promise`, you can use the `isSubmitting` from `Formik` and remove the `isCreatingCollection` state.

```suggestion
  const handleCreate = React.useCallback(async (values) => {
    try {
      await create.mutateAsync(values);
      closeCreateCollectionModal();
      showToast(intl.formatMessage(messages.createCollectionSuccess));
    } catch (error) {
      showToast(intl.formatMessage(messages.createCollectionError));
    }
  }, []);
```

Reply:
Thanks for the recommendation. Updated: https://github.com/openedx/frontend-app-course-authoring/pull/1259/commits/636d2312451f3c3745591292de2ffd510eb40294